views/head.pug

```diff
@@ -32,6 +32,9 @@ style#theme
 // The atob/base64 stuff below is to prevent any XSS attacks: now we have user content in the options (from
 // short links), it's possible to do XSS attacks, by leaving <script> tags in the source etc, which can be
 // interpreted by the browser.
-script window.compilerExplorerOptions = JSON.parse(decodeURIComponent("!{encodeURIComponent(compilerExplorerOptions)}"))
+script(src=`${httpRootDir}client-options.js?hash=${optionsHash}`)
+script
+  | var extraOptions = JSON.parse(decodeURIComponent("!{encodeURIComponent(compilerExplorerOptions)}"));
+  | for (k in extraOptions) { window.compilerExplorerOptions[k] = extraOptions[k]; }
```

Comment:
I think this is ok, but maybe we should keep options from client-options in one a different object, and then merge them into compilerExplorerOptions with `Object.assign({}, baseOptions, extraOptions);` (though I don't know that's worth it, or indeed if object.assign is available in older Javascripts...so...maybe ignore?)

Reply:
it occurs to me that there aren't actually any cases that override anything from the "static" options with values from "extra" - maybe there were in the past?

Maybe it would just be simpler to have `compilerExplorerConfig` with all the static stuff and `compilerExplorerState` for all the per-request variable bits?

Reply:
Yeah that works for me! Clearer too. Doesn't have to be in this PR if it looks like it'll cause trouble